domain/src/main/java/com/example/domain/utils/Results.kt

```diff
@@ -0,0 +1,8 @@
+package com.example.domain.utils
+
+sealed class Results< out T> {
+    data object Loading : Results<Nothing>()
+    data class Success< T>(val data: T) : Results<T>()
+    data class Error(val exception: Throwable) : Results<Nothing>()
+}
```

Comment:
Classes like these has no usage in a domain layer. The `domain` layer is written for `success` cases only. It's the `data` layer that categorises and rethrows errors, and it's the `presentation` layer that handles the foreseen and unforeseen errors.